src/compiler/checker.ts

```diff
@@ -10119,6 +10202,18 @@ namespace ts {
         }
 
         function checkApplicableSignature(node: CallLikeExpression, args: Expression[], signature: Signature, relation: Map<RelationComparisonResult>, excludeArgument: boolean[], reportErrors: boolean) {
+            const headMessage = Diagnostics.Argument_of_type_0_is_not_assignable_to_parameter_of_type_1;

```

Comment:
Yeah, it's not great. Ryan suggested the custom message

'this' context of type '{0}' is not assignable to method 'this' of type '{1}'